File: javascript-leetcode/225.js
/**
 * Initialize your data structure here.
 */
var MyStack = function() {
  this.queue = [];
};

/**
 * Push element x onto stack.
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function(x) {
  this.queue.unshift(x);
};

/**
 * Removes the element on top of the stack and returns that element.
 * @return {number}
 */
MyStack.prototype.pop = function() {
  for (let i = this.queue.length - 1; i > 0; i -= 1) {
    this.queue.unshift(this.queue.pop());
  }
  return this.queue.pop();
};

/**
 * Get the top element.
 * @return {number}
 */
MyStack.prototype.top = function() {
  return this.queue[0];
};

/**
 * Returns whether the stack is empty.
 * @return {boolean}
 */
MyStack.prototype.empty = function() {
  return this.queue.length === 0;
};

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */

File: python-leetcode/225.py
class MyStack:

  def __init__(self):
    """
    Initialize your data structure here.
    """
    self.queue1 = []
    self.queue2 = []

  def push(self, x: int) -> None:
    """
    Push element x onto stack.
    """
    if len(self.queue1) > 0:
      self.queue1.append(x)
    else:
      self.queue2.append(x)

  def pop(self) -> int:
    """
    Removes the element on top of the stack and returns that element.
    """
    queue1 = self.queue1 if len(self.queue1) > 0 else self.queue2
    queue2 = self.queue2 if len(self.queue1) > 0 else self.queue1
    for i in range(len(queue1) - 1):
      queue2.append(queue1.pop(0))
    return queue1.pop(0)

  def top(self) -> int:
    """
    Get the top element.
    """
    queue1 = self.queue1 if len(self.queue1) > 0 else self.queue2
    queue2 = self.queue2 if len(self.queue1) > 0 else self.queue1
    for i in range(len(queue1) - 1):
      queue2.append(queue1.pop(0))
    top = queue1.pop(0)
    queue2.append(top)
    return top

  def empty(self) -> bool:
    """
    Returns whether the stack is empty.
    """
    return len(self.queue1) == 0 and len(self.queue2) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()
